diagnostics/diag.go

```diff
@@ -336,3 +346,16 @@ func (d *Diagnostics) handleNewStream(s inet.Stream) {
 	d.HandleMessage(context.Background(), s)
 	s.Close()
 }
+
+func GetCommitHash() (string, error) {
+	gopath := os.Getenv("GOPATH")
+	ipfs := gopath + "/src/github.com/jbenet/go-ipfs"
```

Comment:
Yeah... the problem is we cant force go generate steps to be included in the `go build` process. And unless we *force* people to use make to build, which sucks, we cant guarantee that the code has the right hash embedded...